src/olympia/scanners/tests/test_admin.py

```diff
@@ -601,3 +609,214 @@ def test_get_fields_for_non_admins(self):
         request.user = user
         assert 'definition' not in self.admin.get_fields(request=request)
         assert 'formatted_definition' in self.admin.get_fields(request=request)
+
+
+class TestScannerQueryRuleAdmin(TestCase):
+    def setUp(self):
+        super().setUp()
+
+        self.user = user_factory()
+        self.grant_permission(self.user, 'Admin:ScannersQuery')
+        self.client.login(email=self.user.email)
+        self.list_url = reverse('admin:scanners_scannerqueryrule_changelist')
+
+    def test_list_view(self):
+        ScannerQueryRule.objects.create(name='bar', scanner=YARA)
+        response = self.client.get(self.list_url)
+        assert response.status_code == 200
+
+    def test_list_view_is_restricted(self):
+        user = user_factory()
+        self.grant_permission(user, 'Admin:Curation')
+        self.client.login(email=user.email)
+        response = self.client.get(self.list_url)
+        assert response.status_code == 403
+
+    def test_change_view_contains_link_to_results(self):
+        rule = ScannerQueryRule.objects.create(name='bar', scanner=YARA)
+        result = ScannerQueryResult(scanner=YARA)
+        result.add_yara_result(rule=rule.name)
+        result.save()
+        ScannerQueryResult.objects.create(scanner=YARA)  # Doesn't match
```

Comment:
I don't understand what this is doing, nor the comment `Doesn't match`.

Reply:
It adds an extra result that doesn't match the rule. Below, we check how many results are linked to that rule, so if we found `2` that would mean our extra result was taken into account, which would be a bug.